server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/liquidbitcoin/LiquidBitcoinExtension.java

```diff
@@ -0,0 +1,128 @@
+    private static final ICryptoCurrencyDefinition DEFINITION = new LiquidBitcoinDefinition();
+
+    @Override
+    public String getName() {
+        return "BATM Liquid Network Bitcoin extension";
+    }
+
+    @Override
+    public IWallet createWallet(String walletLogin, String tunnelPassword) {
+        try {
+            if (walletLogin != null && !walletLogin.trim().isEmpty()) {
+                StringTokenizer st = new StringTokenizer(walletLogin, ":");
+                String walletType = st.nextToken();
+
+                if ("elementsdbtcnoforward".equalsIgnoreCase(walletType)) {
+                    //"elementsdbtcnoforward:protocol:user:password:ip:port:walletname"
+
+                    String protocol = st.nextToken();
+                    String username = st.nextToken();
+                    String password = st.nextToken();
+                    String hostname = st.nextToken();
+                    int port = Integer.parseInt(st.nextToken());
+                    String walletName = "";
+                    if (st.hasMoreTokens()) {
+                        walletName = st.nextToken();
+                    }
+
+                    if (ctx != null) {
+                        InetSocketAddress tunnelAddress = ctx.getTunnelManager().connectIfNeeded(walletLogin, tunnelPassword, InetSocketAddress.createUnresolved(hostname, port));
+                        hostname = tunnelAddress.getHostString();
+                        port = tunnelAddress.getPort();
+                    }
+
+                    if (protocol != null && username != null && password != null && hostname != null && walletName != null) {
+                        String rpcURL = protocol + "://" + username + ":" + password + "@" + hostname + ":" + port;
+                        return new ElementsdRPCWalletWithUniqueAddresses(rpcURL, walletName);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            ExtensionsUtil.logExtensionParamsException("createWallet", getClass().getSimpleName(), walletLogin, e);
+        }
+        return null;
+    }
+
+    @Override
+    public ICryptoAddressValidator createAddressValidator(String cryptoCurrency) {
+        if (CryptoCurrency.L_BTC.getCode().equalsIgnoreCase(cryptoCurrency)) {
+            return new LiquidBitcoinAddressValidator();
+        }
+        return null;
+    }
+
+    @Override
+    public IRateSource createRateSource(String sourceLogin) {
+        if (sourceLogin != null && !sourceLogin.trim().isEmpty()) {
+            try {
+                StringTokenizer st = new StringTokenizer(sourceLogin, ":");
+                String exchangeType = st.nextToken();
+
+                if ("l_btcfix".equalsIgnoreCase(exchangeType)) {
+                    BigDecimal rate = BigDecimal.ZERO;
+                    if (st.hasMoreTokens()) {
+                        try {
+                            rate = new BigDecimal(st.nextToken());
+                        } catch (Throwable e) {
```

Comment:
Either log the caught exception or rethrow it, but don't consume it like that. If the catch block is empty, we won't know that something is wrong.